private/protocol/json/jsonutil/build.go

```diff
@@ -233,6 +234,12 @@ func buildScalar(v reflect.Value, buf *bytes.Buffer, tag reflect.StructTag) erro
 				}
 				buf.WriteByte('"')
 			}
+		case aws.JSONValue:
+			str, err := protocol.EncodeJSONValue(converted, protocol.QuotedEscape)
+			if err != nil {
+				return fmt.Errorf("unable to encode JSONValue, %v", err)
+			}
+			buf.WriteString(str)
```

Comment:
I just realized we squash all these error. Is that something we want to do?

Reply:
definitely want to return errors from these functions. If the protocol marshaler is squashing them then those marshalers should be updated not to squash the errors.

Reply:
Ah you mean errors on write, sorry missed that.  Those should be captured... I don't think the SDK does that for any writes currently like this since the type is a bytes.buffer.